### DISCUSSION TITLE
pattern string escape all the string.

### DISCUSSION THREAD

**Discussion by jianhe-fun** [BurntSushi/ripgrep#2545] [General] [answered]
in postgres, you can $$  this is your string $$.
between $$, except $, what you enter is really what is, literal string. You don't need escape special character.

in regex there are so many special characters. Would be better to escape it once for all.
is there any way to do it in rg. so I don't need think about special character.

**Comment by BurntSushi** [comment, marked as answer]
Yes. The `-F/--fixed-strings` flag. For example, `rg -F '+'` is equivalent to `rg '\+'`.

**Reply by jianhe-fun** [reply]
then the following should not yield any error, since all special character already escaped, if I specified `--fixed-strings`

rg --fixed-strings 'substring(ln,'(?:Execution Time:\s)([0-9.]*)')'

**Reply by BurntSushi** [reply]
Right. It won't return an error. The string will be searched as a literal. So for example, `(?:` will literally match `(?:`.

**Reply by jianhe-fun** [reply]
in postgres world, would be 
rg  $$substring(ln,'(?:Execution Time:\s)([0-9.]*)')$$. 
it will match the literal string: substring(ln,'(?:Execution Time:\s)([0-9.]*)')

**Reply by BurntSushi** [reply]
Okay? Are you asking a question?
